### PR TITLE
Improve mention visibility by adding a background color

### PR DESCRIPTION
**Pull request by RemcoSmitsDev**
When the chat if going fast, It's hard to see who is mentioning you, so this feature will make it more clear by the UI instead of needing to read all the messages.

Screenshot 2024-01-29 at 21 19 07

Release Notes:

- Added background to messages that mention you.

### CODE REVIEW

File: crates/collab_ui/src/chat_panel.rs [hunk 1]
```diff
@@ -335,67 +336,77 @@ impl ChatPanel {
...
-            .when(!is_continuation_from_previous, |this| {
-                this.pt_3().child(
-                    h_flex()
-                        .text_ui_sm()
-                        .child(
-                            div().absolute().child(
+        let mentioning_you = message
+            .mentions
+            .iter()
+            .any(|m| m.1 == self.client.user_id().unwrap());
```
Comment (maxbrunsfeld) [inline crates/collab_ui/src/chat_panel.rs:342]: This use of `.unwrap()` is most likely fine, but our ideal policy is to only panic in code paths that you can tell are unreachable *locally* (i.e. based on the immediate surrounding method).

Could you make this slightly more defensive as follows:

```suggestion
            .any(|m| Some(m.1) == self.client.user_id());
```

File: crates/collab_ui/src/chat_panel.rs [hunk 2]
```diff
@@ -335,67 +336,77 @@ impl ChatPanel {
...
+
+        v_flex().w_full().relative().child(
+            div()
+                .bg(if mentioning_you {
+                    Fill::from(cx.theme().colors().background)
+                } else {
+                    Fill::default()
+                })
+                .p_1()
+                .rounded_md()
```
Comment (maxbrunsfeld) [inline crates/collab_ui/src/chat_panel.rs:352]: Just curious - how does this look when two consecutive message both mention you?

**Review by maxbrunsfeld — CHANGES_REQUESTED** [review]
This is a great improvement. Thanks for including the screenshot. I left one small code suggestion, and a design question.

File: crates/collab_ui/src/chat_panel.rs (hunk 1 above)
Comment (RemcoSmitsDev) [reply crates/collab_ui/src/chat_panel.rs:342]: Yeah, makes sense. Didn't know you could fix it like this, I'm kinda new to Rust.

File: crates/collab_ui/src/chat_panel.rs (hunk 2 above)
Comment (RemcoSmitsDev) [reply crates/collab_ui/src/chat_panel.rs:352]: Now it looks like this:
Screenshot 2024-01-29 at 22 30 38

Comment (RemcoSmitsDev) [reply crates/collab_ui/src/chat_panel.rs:352]: Haven't tested that one:
Screenshot 2024-01-29 at 22 33 52

When there is a message in between, it looks nice:
Screenshot 2024-01-29 at 22 33 05

Comment (RemcoSmitsDev) [reply crates/collab_ui/src/chat_panel.rs:352]: I changed it locally to this. But one note, this brings a little inconsistency with spacing between messages that do/not have a mention inside the message. It's not much, but I'm not sure if this is a problem.
Screenshot 2024-01-29 at 22 39 46

Comment (RemcoSmitsDev) [reply crates/collab_ui/src/chat_panel.rs:351]: I fixed that inconsistent spacing issue in my latest commit.

**Comment by maxbrunsfeld** [comment]
Thanks @RemcoSmitsDev; this will ship in Zed 0.121.

**Pull request merged**